### PR TITLE
fix: Adapt the menubar to the primary color if a custom backgorund image has been set

### PR DESCRIPTION
**Pull request by juliusknorr**
## Steps to reproduce:

- Set white as a primary color
- Set a bright background image as the admin
- Disable user theming (personal background image)

## Before

App icons where white on white

## After

App icons properly adapt to the primary color

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]
Could you also add a use case for this in the cypress tests please?

**Review by szaimen — CHANGES_REQUESTED** [review]
This doesnt seem to work if you choose a dark primary color. Also the app title in the heading doesnt seem to be inverted correctly...

**Comment by skjnldsv** [comment]
We kinda had that discussion with https://github.com/nextcloud/server/issues/35482
If we want to apply a different rule following the primary, then we need to apply some overriding variables to the header element

**Comment by juliushaertl** [comment]
Won't have time looking into that further this week, so If anyone wants to jump in feel free. Otherwise I'll check again somewhen next week.

**Comment by MichaIng** [comment]
Related: #37379
Avatar and status icon get inverted as well, so it must be taken care that only app icons are inverted.

File: apps/theming/lib/Themes/CommonThemeTrait.php [hunk 1]
```diff
@@ -101,6 +101,11 @@ protected function generateGlobalBackgroundVariables(): array {
 			$variables['--background-image-invert-if-bright'] = $isDefaultPrimaryBright ? 'invert(100%)' : 'no';
 		}
 
+		// Adapt the menubar to the primary color if a custom backgorund image has been set
```
Comment (skjnldsv) [inline apps/theming/lib/Themes/CommonThemeTrait.php:104]:
```suggestion
		// Adapt the menubar to the primary color if a custom background image has been set
```

**Review by skjnldsv — APPROVED** [review]
One tiny comment, but all good otherwise

**Comment by szaimen** [comment]
@skjnldsv the PR hasnt changed since I reviewed it so what shall I review here?

**Comment by juliushaertl** [comment]
I revived this and pushed a cypress test.

> This doesnt seem to work if you choose a dark primary color. Also the app title in the heading doesnt seem to be inverted correctly...

I could not reproduce this anymore

**Comment by juliushaertl** [comment]
Thinking more about it i don't think this is the right approach, as long as we have no detection or setting for if the background is dark or bright, we can only make it work in either combination. The primary color is only a good indicator for inverting app icons if no background is used. Will close this.

**Pull request closed without merging**